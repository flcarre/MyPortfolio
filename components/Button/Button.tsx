import { FunctionComponent, MouseEventHandler } from "react";

type Props = {
  text: string;
  onClick?: MouseEventHandler;
};

const Button: FunctionComponent<Props> = ({ text, onClick }) => (
  <button
    onClick={onClick}
    className="rounded shadow-md flex items-center shadow bg-blue-500 px-4 py-2 text-white hover:bg-blue-600">
    {text}
  </button>
);

export default Button;
