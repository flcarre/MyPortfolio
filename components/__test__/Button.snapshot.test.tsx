import React from "react";
import Button from "../Button/Button";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Button text="Some Text" onClick={() => { }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
