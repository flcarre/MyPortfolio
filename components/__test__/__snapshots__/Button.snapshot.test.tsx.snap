// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="rounded shadow-md flex items-center shadow bg-blue-500 px-4 py-2 text-white hover:bg-blue-600"
  onClick={[Function]}
>
  Some Text
</button>
`;
